src/systemctl/systemctl-edit.c

```diff
@@ -483,6 +483,18 @@ static int trim_edit_markers(const char *path) {
                 strshorten(contents_start, contents_end - contents_start);
 
         contents_start = strstrip(contents_start);
+        if (*contents_start && !endswith(contents_start, "\n")) {
+                char *tmp = contents_start;
+                if (MALLOC_SIZEOF_SAFE(contents) - (contents_start - contents) - strlen(contents_start) < 2) {
+                        if ((tmp = realloc(contents, size + 1))) {
+                                contents_start = tmp + (contents_start - contents);
+                                contents = tmp;
+                        }
+                }
+
+                if (tmp)
+                        strcat(contents_start, "\n");
```

Comment:
we do not allow strcat() in our codebase. codeql should have refused that.

Reply:
Yes you do:
```
nabijaczleweli@tarta:~/store/code/systemd$ git grep -w strcat
.github/codeql-custom.qls:      - cpp/unsafe-strcat
src/basic/unit-name.c:                strcat(s, suffix);
src/systemctl/systemctl-edit.c:                        strcat(contents_start, "\n");
```

And no it shouldn't've, [the buffer size is checked](https://codeql.github.com/codeql-query-help/cpp/cpp-unsafe-strcat/#example).

I could replace the strcat with inline strcat-but-worse. Nothing is gained, semantics are lost.